Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -294,11 +294,15 @@ class SourceGenerator {
             // lives outside the project base path
             let isOutOfBasePath = !path.absolute().string.contains(project.basePath.absolute().string)
 
+            // whether the given path is a strict parent of the project base path
+            // e.g. foo/bar is a parent of foo/bar/baz, but not foo/baz
+            let isParentOfBasePath = isOutOfBasePath && ((try? path.isParent(of: project.basePath)) == true)
+
             // has no valid parent paths
-            let isRootPath = (isBaseGroup && isOutOfBasePath) || path.parent() == project.basePath
+            let isRootPath = (isBaseGroup && isOutOfBasePath && isParentOfBasePath) || path.parent() == project.basePath
```

Comment:
I think this line can lose the `&& isParentOfBasePath`. 
`isOutOfBasePath` captures the intent and `isRootPath` is used in the line below. Or am I reading it wrong?

Reply:
I don't think it can. For example, for a project in ., ../foo/bar.txt shouldn't be considered the root path -- the root path should be a strict parent of the current directory